setup.py

```diff
@@ -255,8 +259,8 @@ def run(self):
                 EIGEN3_INCLUDE_DIR = os.path.join(EIGEN3_INCLUDE_DIR, os.pardir)
             if not os.path.isdir(EIGEN3_INCLUDE_DIR):
                 raise RuntimeError("Could not find Eigen in EIGEN3_INCLUDE_DIR={}. If doing manual install, please set the EIGEN3_INCLUDE_DIR variable with the absolute path to Eigen manually. If doing install via pip, please file an issue at the github site.".format(EIGEN3_INCLUDE_DIR))
+            BUILT_EXTENSIONS = False  # already build repo
```

Comment:
The idea in setting `BUILT_EXTENSIONS` to `True` even if `--skip-build` is given is that in that case we assume everything has already been built. This causes the `error: [Errno 2] No such file or directory: 'build'` failure on Travis CI.

Reply:
But if I am building everything externally say using VS on Windows, the code shouldn't be running the Build extensions code. Would you add a bit more clarification on this ?

Reply:
@danielhers looking at the travis file it looks like the following line should be edited to fix the issue

https://github.com/clab/dynet/blob/feb01a17560eeafb8c5fdb0a39634457ef62d5fa/.travis.yml#L92

It has
```
      python ../../setup.py build --build-dir=.. --skip-build install --user || travis_terminate 1;

```

If I am skipping the build then it should not be required to pass the build argument to setup.py. 

What do you think? Without using that piece of code, the build fails on Windows as the setup.py tries to compile the files using g++. 

If I keep the `BUILD_EXTENSIONS` line as before then I get the error described in #830